frontend/src/views/Organizations/Organizations.jsx

```diff
@@ -21,98 +21,110 @@ import { LinkRenderer } from '@/utils/grid/cellRenderers.jsx'
+      cellRenderer: LinkRenderer,
+      cellRendererParams: {
+        url: (data) => data.data.organizationId
+      }
+    }),
+    []
+  )
 
-  const navigate = useNavigate()
-  const location = useLocation()
+  // Sorting
+  const defaultSortModel = useMemo(
+    () => [{ field: 'name', direction: 'asc' }],
+    []
+  )
 
-  const defaultSortModel = [{ field: 'name', direction: 'asc' }]
+  // For alert messages
+  const [alertMessage, setAlertMessage] = useState('')
+  const [alertSeverity, setAlertSeverity] = useState('info')
 
+  // For downloads
   const apiService = useApiService()
   const [isDownloadingOrgs, setIsDownloadingOrgs] = useState(false)
   const [isDownloadingUsers, setIsDownloadingUsers] = useState(false)
-  const [alertMessage, setAlertMessage] = useState('')
-  const [alertSeverity, setAlertSeverity] = useState('info')
+
+  // For clearing filters
   const [resetGridFn, setResetGridFn] = useState(null)
+  const handleSetResetGrid = useCallback(
+    (fn) => {
+      if (resetGridFn !== fn) {
+        setResetGridFn(() => fn)
+      }
+    },
+    [resetGridFn]
+  )
+
+  const handleClearFilters = useCallback(() => {
+    if (resetGridFn) {
+      resetGridFn()
+    }
+  }, [resetGridFn])
 
+  // Router navigation
+  const navigate = useNavigate()
+  const location = useLocation()
   useEffect(() => {
     if (location.state?.message) {
       setAlertMessage(location.state.message)
       setAlertSeverity(location.state.severity || 'info')
     }
   }, [location.state])
 
+  // Download handlers
   const handleDownloadOrgs = async () => {
     setIsDownloadingOrgs(true)
     setAlertMessage('')
     try {
       await apiService.download('/organizations/export')
-      setIsDownloadingOrgs(false)
     } catch (error) {
       console.error('Error downloading organization information:', error)
-      setIsDownloadingOrgs(false)
       setAlertMessage(t('org:orgDownloadFailMsg'))
       setAlertSeverity('error')
+    } finally {
+      setIsDownloadingOrgs(false)
     }
   }
 
   const handleDownloadUsers = async () => {
     setIsDownloadingUsers(true)
     try {
       await apiService.download(apiRoutes.exportUsers)
-      setIsDownloadingUsers(false)
     } catch (error) {
       console.error('Error downloading user information:', error)
-      setIsDownloadingUsers(false)
       setAlertMessage(t('org:userDownloadFailMsg'))
       setAlertSeverity('error')
+    } finally {
+      setIsDownloadingUsers(false)
     }
   }
 
-  const defaultColDef = useMemo(
-    () => ({
-      cellRenderer: LinkRenderer,
-      cellRendererParams: {
-        url: (data) => data.data.organizationId
-      }
-    }),
-    []
-  )
-
-  const handleSetResetGrid = useCallback((fn) => {
-    setResetGridFn(() => fn)
-  }, [])
-
-  const handleClearFilters = useCallback(() => {
-    if (resetGridFn) {
-      resetGridFn()
-    }
-  }, [resetGridFn])
+  // Row ID
+  const getRowId = useCallback((params) => params.data.organizationId, [])
```

Comment:
getRowId should return a string to prevent warnings from ag-grid. Is this throwing warnings?

Reply:
no warnings shown but I have hard casted this just in case.